File: 405_Convert_a_Number_to_Hexadecimal/bit_operation_solution.cpp
class Solution {
public:
    string toHex(int num) {
        if (num == 0) {
            return "0";
        }
        string result = "";
        for (int i = 7; i >= 0; --i) {
            int val = (num >> (4 * i)) & 0xf;
            if (result.size() > 0 || val) {
                result += val < 10? char(val + '0') : char('a' + val - 10);
            }
        }
        return result;
    }
};

File: 137_Single_Number_II/bit_operation_solution.py
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        result = 0
        for i in range(0, 31):
            total = sum(((num >> i) & 1) for num in nums)
            if total % 3:
                result |= (1 << i)
                temp = bin(result)
        total = sum(((num >> 31) & 1) for num in nums) 
        if total % 3:
            # result |= (1 << 31)
            result -= (1 << 31)
            temp = bin(result)
        return result